Yoast/javascript
# P2-215 Create Select instruction.

## Summary


This PR can be summarized in the following changelog entry:

* [schema-blocks] Creates a Select block instruction to the schema-blocks package. With this instruction, a `select` element can be added to a schema template. 

## Relevant technical choices:

* Decided to make this into a single `select` element, so multiple values cannot be selected currently.
	* The issues and designs of the schema templates currently do not call for `select`s that expect multiple values.
	* I would propose, if we need one, to add it as a separate instruction when needed.
		* There are fundamental differences in how the values are stored (multiple instead of one) and in the design of the `select` element. 
* The `createElement` import is now ignored for the `no-unused-vars` ESLint warning.
	* `createElement` is needed to create React elements, but is never called directly when using JSX. This leads to a false positive in ESLint. 		
* I adapted the transform configuration in the Jest configuration so Jest is able to parse `tsx` files. 	 

## Test instructions

This PR can be tested by following these steps:

* In WordPress SEO, in a schema template (for example `src/schema-templates/recipe.block.php`), add a `Select` instruction.
	* E.g. like this:
	```
	{{select label="Cuisine" name="cuisine" options={ "Caribbean": "caribbean", "German": "german", "Cantonese": "cantonese" } }}
	``` 
* Build the plugin as you normally would.
	* Make sure that your monorepo is linked and on the right branch. 
	* Make sure that you have built the `schema-blocks` package using `yarn build`. 	
	* Make sure that you run `grunt build:dev` in the plugin, so the changes from the `schema-blocks` are integrated within the JavaScript files of the plugin.  	
* Create a new post.
* Add a recipe block.
* See that it contains your added `select` instruction in the form of a drop down box with a label.
* Select an option in the drop down box.
* Save the post.
* View the post on the frontend of your website.
* See that it shows the label and selected value in the recipe.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  * N/A: this PR only affects the `schema-blocks` package, which is not yet released and behind a feature flag.

## UI changes

## Quality assurance


Fixes #
